### PR TITLE
[ios] unused calls

### LINKED ISSUE #405: Many 'Result of call to x is unused' warnings
Unused function results are marked as warnings in Swift. Many of these are unhandled errors. We should also consider throwing instead of returning a `Bool` that indicate success or failure.

### PR DESCRIPTION
**Pull request by jacobbullock**
fixes #405

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift [hunk 1]
```diff
@@ -191,20 +191,23 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
...
     return false
   }
+  
 
   /// Set the current keyboard.
   ///
-  /// - Returns: Whether the keyboard was set successfully
-  public func setKeyboard(_ kb: InstallableKeyboard) -> Bool {
+  /// - Throws: error if the keyboard was unchanged
+  public func setKeyboard(_ kb: InstallableKeyboard) throws {
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/Classes/Manager.swift]: Is this a public API? If so, we can't change the behaviour.

It seems sensible to allow a no-op `setKeyboard`? Is this just to resolve unused result warnings?

**Comment by jahorton** [comment]
@gabrielwong had originally intended for those values to be preserved and checked for error handling in some fashion, so I know the `throws` was far closer to the original intent for his work on the iOS conversion.  I'd initially suggested the `_ =` syntax in #409 back in December, which ended up being outright closed.

**Comment by jacobbullock** [comment]
@mcdurdin @jahorton is there a consensus on how you want me to move forward with this?  I can make the change so we can get it in for this sprint

**Comment by mcdurdin** [comment]
There are several apps that use Keyman Engine so I'd like to keep existing behaviour.

**Comment by jacobbullock** [comment]
@mcdurdin I have made the update for the `setKeyboard` calls, can you check this one?

**Pull request merged**